### PR TITLE
GLT-3144: allow configuring a secondary naming scheme

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: miso-web/src/main/webapp/scripts/form_project.js [hunk 1]
```diff
@@ -94,6 +101,18 @@ FormTarget.project = (function($) {
...
+          data: 'secondaryNaming',
+          type: 'checkbox',
+          include: !!config.naming.secondary,
+          onChange: function(newValue, form) {
+            var scheme = config.naming[newValue ? 'secondary' : 'primary'];
+            form.updateField('shortName', {
+              required: scheme.shortNameRequired,
+              disabled: !scheme.shortNameModifiable
+            });
+          }
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/form_project.js]: Would it make sense to move this before the "Short Name" column, as that'll be affected by changes to this checkbox (and could lead to users having previously-valid data be marked invalid, or have to enter in invalid data which will then be validated by checking this box as they fill in left-to-right)?

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -4,6 +4,9 @@ Changes:
 
   * Added configuration for library aliquot alias generator and validator
   * Added V2 naming scheme
```
Comment (callunity) [inline RELEASE_NOTES.md]: Might be worth changing this to `v2` to match the case expected in `miso.properties`

File: miso-web/src/main/webapp/scripts/form_project.js (hunk 1 above)
Comment (djcooke) [reply miso-web/src/main/webapp/scripts/form_project.js]: Yeah, makes sense. Will move

**Comment by djcooke** [comment]
@callunity: My intention was that when we switch over to using the V2 scheme for all new projects, we'd flip them so that V2 is configured as the primary scheme and OICR is secondary (flipping the bit on all projects at the same time of course)

**Comment by callunity** [comment]
@djcooke that makes sense. It could be helpful to others to add that suggestion to the baremetal-installation-guide in case someone else goes the same route but doesn't think of that possibility (but I think the chances of that happening are realistically pretty small).

File: RELEASE_NOTES.md (hunk 2 above)
Comment (djcooke) [reply RELEASE_NOTES.md]: Changed

File: miso-web/src/main/webapp/scripts/form_project.js (hunk 1 above)
Comment (djcooke) [reply miso-web/src/main/webapp/scripts/form_project.js]: Moved

**Comment by djcooke** [comment]
@callunity: added to the guide

File: RELEASE_NOTES.md (hunk 2 above)
Comment (callunity) [reply RELEASE_NOTES.md]: Looks good to me

**Pull request merged**